core/src/test/scala/org/apache/spark/deploy/SparkSubmitSuite.scala

```diff
@@ -482,7 +482,7 @@ class SparkSubmitSuite
     val (childArgs, classpath, conf, mainClass) = submit.prepareSubmitEnvironment(appArgs)
 
     val childArgsMap = childArgs.grouped(2).map(a => a(0) -> a(1)).toMap
-    childArgsMap.get("--primary-java-resource") should be (Some("file:/home/thejar.jar"))
+    childArgsMap.get("--primary-java-resource").get should endWith ("file:/home/thejar.jar")
```

Comment:
Given your description, this doesn't seem correct. It won't end with a string that starts with `file:`.

Reply:
Like @srowen mentioned, this PR fails still.
```
[info] - handles k8s cluster mode *** FAILED *** (534 milliseconds)
[info]   "file:/System/Volumes/Data/home/thejar.jar" did not end with substring "file:/home/thejar.jar" (SparkSubmitSuite.scala:485)
```

Reply:
@toujours33 . You may want to try this.
```scala
childArgsMap.get("--primary-java-resource").get should endWith ("/home/thejar.jar")
```

Reply:
Sorry, i commited the wrong code. 
Thanks for your correcting, will fix it soon

Reply:
@toujours33 We need to check if there are other failed cases like this

Reply:
@LuciferYang  For sure, I'm running all cases on MacOs to check if there are other failed cases like this. Before that, this pr will temporarily marked as draft.